observelib/observe/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java

```diff
@@ -153,7 +156,10 @@ public boolean addTag(String tagKey, String tagValue, long spanId, Strand strand
         if (!enabled) {
             return false;
         }
-        ObserverContext observerContext = observerContextList.get(spanId);
+        ObserverContext observerContext = null;
+        if (observerContextList.containsKey(spanId)) {
+            observerContext = observerContextList.get(spanId);
+        }
```

Comment:
```suggestion
        ObserverContext observerContext = observerContextList.getOrDefault(spanId, null);
```

Reply:
Do we need to check contains here. Next line we check value not null. So just observerContext = observerContextList.get(spanId); should be enough

Reply:
@chiranSachintha addressed review suggestions

Reply:
@warunalakshitha If it is not null, it will throw an error while trying to get object with a key which is not present in the HashMap. Should I keep that?